src/inmanta/server/server.py

```diff
@@ -805,14 +806,36 @@ def get_resource_increment_for_agent(self, env: Environment, agent: str) -> Dict
 
         version = cm.version
 
-        increment_ids = self._increment_cache.get(env.id, None)
-        if increment_ids is None:
+        increment = self._increment_cache.get(env.id, None)
+        if increment is None:
             with (yield self._increment_cache_locks[env.id].acquire()):
-                increment_ids = self._increment_cache.get(env.id)
-                if increment_ids is None:
-                    increment_ids_list = yield cm.get_increment()
-                    increment_ids = set(increment_ids_list)
-                    self._increment_cache[env.id] = increment_ids
+                increment = self._increment_cache.get(env.id)
```

Comment:
Why is the default value None not specified here?

Reply:
good catch!